src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -293,7 +294,7 @@ private void WorkerLoop()
                         // Wait for more things to do.
                         bool isWakeupRequestedPipeActive;
                         bool isTimeout;
-                        ThrowIfCURLMError(Interop.Http.MultiWait(multiHandle, _wakeupRequestedPipeFd, out isWakeupRequestedPipeActive, out isTimeout));
+                        ThrowIfCURLMError(Interop.Http.MultiWait(multiHandle, (int)_wakeupRequestedPipeFd.DangerousGetHandle(), out isWakeupRequestedPipeActive, out isTimeout));
```

Comment:
I was trying to cut the changes off at pal_io in System.Native, as this change got bigger than I expected. But looking at it now, this would be a pretty easy change to make.

The other place that is left undone is the sockets code - it is still using int32_t for the socket file descriptor.  I can fix that in a subsequent PR if people feel it would be beneficial.   @pgavlin ???